drivers/I2C.h

```diff
@@ -164,11 +169,17 @@ class I2C : private NonCopyable<I2C> {
 
     /** Acquire exclusive access to this I2C bus
      */
-    virtual void lock(void);
+    virtual void lock(void)
+    {
+        _mutex->lock();
+    }
 
     /** Release exclusive access to this I2C bus
      */
-    virtual void unlock(void);
+    virtual void unlock(void)
+    {
+        _mutex->unlock();
+    }
```

Comment:
The implementation for lock and unlock was in .cpp. It should be left there.

Reply:
See my overall comment here: https://github.com/ARMmbed/mbed-os/pull/10955#issuecomment-509937556